task9/script.js

```diff
@@ -0,0 +1,433 @@
+                return true;
+            }
+            return false;
+        }
+
+        edit(id, post) {
+            var ind = this.getInd(id);
+            var tmpPost = Object.assign({}, this._posts[ind]);
+            if (typeof post.description !== 'undefined') {
+                tmpPost.description = post.description;
+            }
+            if (typeof post.photoLink !== 'undefined') {
+                tmpPost.photoLink = post.photoLink;
+            }
+            if (this._validate(tmpPost)) {
+                this._posts[ind] = tmpPost;
+                return true;
+            }
+            return false;
+
+        }
+
+
+        remove(id) {
+            var ind = this.getInd(id);
+            this._posts.splice(ind, 1);
+        }
+
+
+        getPage(skip = 0, top = 10, filterConfig) {
+            var arr = [];
+            if (typeof (filterConfig) !== 'undefined') {
+                var toDate = filterConfig.to;
+                var fromDate = filterConfig.from;
+                var author = filterConfig.author;
+            }
+            var skipCount = 0;
+            var topCount = 0;
+            if (typeof (toDate) === 'undefined') {
+                toDate = this._posts[this._posts.length - 1].createdAt;
+            }
+            if (typeof (fromDate) === 'undefined') {
+                fromDate = this._posts[0].createdAt;
+            }
+
+
+            if (typeof (author) === 'string' && author !== '') {
+                for (var i = this._posts.length - 1; i >= 0; i--) {
+                    if (skipCount >= skip && topCount < top && fromDate <= this._posts[i].createdAt && toDate >= this._posts[i].createdAt && author === this._posts[i].author) {
+                        arr.push(this._posts[i]);
+                        topCount++;
+                    }
+                    if (topCount >= top) {
+                        break;
+                    }
+                    skipCount++;
+                }
+            } else if (typeof (author) !== 'string' || author === '') {
+                for (var i = this._posts.length - 1; i >= 0; i--) {
+                    if (skipCount >= skip && topCount < top && fromDate <= this._posts[i].createdAt && toDate >= this._posts[i].createdAt) {
+                        arr.push(this._posts[i]);
+                        topCount++;
+                    }
+                    if (topCount >= top) {
+                        break;
+                    }
+                    skipCount++;
+                }
+            }
+
+            return arr;
+
+        }
+
+        addAll(posts) {
+            var arr = [];
+            for (let i = 0; i < posts.length; i++) {
+                if (this._validate(posts[i])) {
+                    this._posts.push(posts[i]);
+                } else {
+                    arr.push(posts[i]);
+                }
+            }
+            return arr;
+        }
+
+
+    }
+
+    class view {
+
+        static build(post) {
```

Comment:
You can try to use template strings and create some templates instead of setting all the required props manually, so it will look better. You can find an example [here](https://wesbos.com/template-strings-html)